apps/frontend/src/pages/tasks/libs/components/deadline/deadline.tsx

```diff
@@ -14,55 +14,65 @@ import styles from "./styles.module.css";
 
 type Properties = {
 	deadline: string;
+	onExpire: (() => void) | undefined;
 };
 
-const Deadline: React.FC<Properties> = ({ deadline }: Properties) => {
+const Deadline: React.FC<Properties> = ({ deadline, onExpire }: Properties) => {
 	const [countdown, setCountdown] = useState<Countdown>(COUNTDOWN_EXPIRED);
 	const [isExpired, setIsExpired] = useState<boolean>(false);
+	const onExpireReference = useRef(onExpire);
 
 	const clockIconName = isExpired ? "clockInactive" : "clockActive";
 	const countdownStyleClass = getValidClassNames(
 		styles["countdown"],
 		isExpired && styles["expired"],
 	);
 
-	const calculateDaysUntilDeadline = useCallback((): boolean => {
-		const deadlineTime = new Date(deadline).getTime();
-		const currentTime = Date.now();
-		const timeToDeadline = deadlineTime - currentTime;
-
-		if (timeToDeadline < DEADLINE_OVER) {
-			setCountdown(COUNTDOWN_EXPIRED);
-			setIsExpired(true);
-
-			return true;
-		}
-
-		const days = Math.floor(timeToDeadline / MillisecondsPerUnit.DAY);
-		const hours = Math.floor(
-			(timeToDeadline % MillisecondsPerUnit.DAY) / MillisecondsPerUnit.HOUR,
-		);
-		const minutes = Math.floor(
-			(timeToDeadline % MillisecondsPerUnit.HOUR) / MillisecondsPerUnit.MINUTE,
-		);
-
-		const formattedDays = String(days);
-		const formattedHours = String(hours).padStart(TimePad.HOURS, TIME_PAD_FILL);
-		const formattedMinutes = String(minutes).padStart(
-			TimePad.MINUTES,
-			TIME_PAD_FILL,
-		);
-
-		setCountdown({
-			days: formattedDays,
-			hours: formattedHours,
-			minutes: formattedMinutes,
-		});
-
-		return false;
-	}, [deadline]);
+	useEffect(() => {
+		onExpireReference.current = onExpire;
```

Comment:
Why do we have a reference for a prop?

Reply:
I've tested the work without it and removed it. It doesn't make sense to keep it now.

I had a problem with "Maximum update depth exceeded" and then an issue with the same card appearing in the modal on hot reload. This approach partially helped me, I believe. In the end, the problem was in the calculateDaysUntilDeadline, which was defined using useCallback.